lib/set_game/game_supervisor.ex

```diff
@@ -0,0 +1,38 @@
+defmodule SetGame.GameSupervisor do
+  use DynamicSupervisor
+
+  alias SetGame.GameServer
+
+  def start_game() do
+    name = GameServer.find_available_name() |> GameServer.via_tuple()
+
+    case DynamicSupervisor.start_child(
+           __MODULE__,
+           GameServer.child_spec(name)
+         ) do
+      {:ok, _pid, _info} -> {:ok, name}
+      {:ok, _pid} -> {:ok, name}
+      other -> other
+    end
+  end
```

Comment:
There is a minuscule chance of a race condition here, if the name _somehow_ gets picked up by another server between lines 7 and 9.



```suggestion
  def start_game() do
    # generate_name/0 should be made public
    name = GameServer.generate_name() |> GameServer.via_tuple()

    case DynamicSupervisor.start_child(
           __MODULE__,
           GameServer.child_spec(name)
         ) do
      {:ok, _pid, _info} -> {:ok, name}
      {:ok, _pid} -> {:ok, name}
      
      # Recurse HERE instead
      {:error, {:already_started, _pid}} -> start_game()
      
      other -> other
    end
  end
```

Reply:
Thank you! Applied a slightly different fix that also avoids exposing the generate name function!